lib/connection.js

```diff
@@ -112,15 +112,24 @@ class Connection extends EventEmitter {
   }
 
   // Send a handshake packet over the connection
-  //   appName - name of an application to connect to
+  //   app - string or object, application to connect to as name
+  //         or { name, version }
   //   login - user name (optional)
   //   password - user password (optional)
   //   callback - callback function to invoke after the handshake is completed
   //
-  handshake(appName, login, password, callback) {
+  handshake(app, login, password, callback) {
+    let target;
+    if (typeof app === 'string') {
+      target = app;
+    } else if (!app.version) {
```

Comment:
You should also add a possibility to provide `app` as a string containing both name and version like that: `appName@1.0.0`.